hail/python/hailtop/aiogoogle/container/client.py

```diff
@@ -0,0 +1,64 @@
+import hailtop.aiogoogle.auth as google_auth
+
+
+class Client:
+    def __init__(self, project, *, session=None, **kwargs):
+        self._project = project
+        if session is None:
+            session = google_auth.Session(**kwargs)
+        self._session = session
+
+    # list images:
+    # GET /tags/list
+    # returns:
+    # {
+    #   'child': [<image>, ...],
+    #   'manifest': {},
+    #   'name': <project>,
+    #   'tags': []
+    # }
+
+    # list image tags:
+    # GET /{image}/tags/list
+    # returns:
+    # {
+    #   'child': [],
+    #   'manifest': {<diget>: {
+    #       'imageSizeBytes': <size>,
+    #       'layerId': '',
+    #       'mediaType': 'application/vnd.docker.distribution.manifest.v2+json',
+    #       'tag': [<tag>, ...],
+    #       'timeCreatedMs': '<time-in-ms>',
+    #       'timeUploadedMs': '<time-in-ms>'
+    #     }, ...
+    #   },
+    #   'name': '<project>/<image>',
+    #   'tags': [<tag>, ...]
+    # }
+
+    async def get(self, path, **kwargs):
```

Comment:
This is inspired by the design of the gidgethub, the Github client we use in CI.  The goal is to make the client lightweight and just handle the protocol for communiating, and not have it know every operation we might want to call.   So instead of saying `client.list_images(image)`, we say `client.get(f'/{image}/tags/list')`.

Other clients might want to handle things like pagination (logging and compute APIs), or treating message bodies as file streams (storage API).